Source/AwsCommonRuntimeKit/mqtt/Mqtt5Packets.swift

```diff
@@ -179,63 +179,105 @@ public class PublishPacket: CStruct {
 
-                        if let _subscriptionPointer = subscriptionPointer {
-                            raw_publish_view.subscription_identifiers = _subscriptionPointer
+                        if let subscriptionPointer = subscriptionPointer {
+                            raw_publish_view.subscription_identifiers = subscriptionPointer
                             raw_publish_view.subscription_identifier_count = subscriptionIdentifiers!.count
                         }
 
                         return withOptionalUserPropertyArray(
                             of: userProperties) { userPropertyPointer in
-                            if let _userPropertyPointer = userPropertyPointer {
+                            if let userPropertyPointer = userPropertyPointer {
                                 raw_publish_view.user_property_count = userProperties!.count
                                 raw_publish_view.user_properties =
-                                    UnsafePointer<aws_mqtt5_user_property>(_userPropertyPointer)
+                                    UnsafePointer<aws_mqtt5_user_property>(userPropertyPointer)
                             }
-                            return withOptionalByteCursorPointerFromString(
+                            return withOptionalByteCursorPointerFromStrings(
                             responseTopic,
-                            correlationData,
-                            contentType) { cResponseTopic, cCorrelationData, cContentType in
+                            contentType) { cResponseTopic, cContentType in
                                 raw_publish_view.content_type = cContentType
-                                raw_publish_view.correlation_data = cCorrelationData
                                 raw_publish_view.response_topic = cResponseTopic
 
-                                return body(raw_publish_view)
+                                return withOptionalByteCursorPointerFromData(to: self.correlationData) {  cCorrelationData in
+                                    raw_publish_view.correlation_data = cCorrelationData
+                                    return body(raw_publish_view)
+                                }
                             }
                         }
                     }
                 }
             }
         }
     }
+
+    static func convertFromNative(_ from: UnsafePointer<aws_mqtt5_packet_publish_view>?) -> PublishPacket? {
+        guard let from = from else {
+            return nil
+        }
+        let publishView = from.pointee
+
+        let payload: Data = Data(bytes: publishView.payload.ptr, count: publishView.payload.len)
+
+        let payloadFormatIndicator: PayloadFormatIndicator? = publishView.payload_format != nil ?
+        PayloadFormatIndicator(rawValue: Int(publishView.payload_format.pointee.rawValue)) : nil
+
+        let messageExpiryInterval = convertOptionalUInt32(publishView.message_expiry_interval_seconds)
```

Comment:
Not relevant to this PR, but I would prefer that we get rid of all the convert* (either removing them or refactoring them) by the final merge. We should avoid introducing new terminologies. For example, if we need a function like this, it should be an extension over Optional, where Element == UInt32 instead of a separate function.